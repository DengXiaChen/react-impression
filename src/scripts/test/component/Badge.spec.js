import React from 'react';
import { shallow } from 'enzyme';
import { Badge } from '../../components/impression';

const setup = () => {
    const props = {
        theme: 'primary',
        content: 'badge test',
        type: 'legend',
        size: 'lg',
    };

    const wrapper = shallow(<Badge {...props} />);

    return {
        props,
        wrapper,
    };
};

describe('Badge', () => {
    const { props, wrapper } = setup();

    // 通过 input 是否存在来判断 Todo组件是否被渲染
    it('render', () => {
        expect(wrapper.find('.badge').exists()).toBe(true);
    });

    it('theme', () => {
        expect(wrapper.find('.bg-primary').exists()).toBe(true);
    });

    it('size', () => {
        expect(wrapper.find('.badge-addon-lg').exists()).toBe(true);
    });

    it('type', () => {
        expect(wrapper.find('.badge-legend').exists()).toBe(true);
    });

    it('content', () => {
        expect(wrapper.find('.badge').text()).toBe(props.content);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
