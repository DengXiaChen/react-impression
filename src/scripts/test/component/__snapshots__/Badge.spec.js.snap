// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="badge badge-legend"
>
    <div
        className="badge-addon badge-addon-lg bg-primary"
    >
        badge test
    </div>
</span>,
  "nodes": Array [
    <span
      className="badge badge-legend"
>
      <div
            className="badge-addon badge-addon-lg bg-primary"
      >
            badge test
      </div>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Badge
        content="badge test"
        size="lg"
        theme="primary"
        type="legend"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Badge {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "content": "badge test",
          "size": "lg",
          "theme": "primary",
          "type": "legend",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="badge badge-legend"
>
          <div
                    className="badge-addon badge-addon-lg bg-primary"
          >
                    badge test
          </div>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="badge badge-legend"
>
          <div
                    className="badge-addon badge-addon-lg bg-primary"
          >
                    badge test
          </div>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Badge
    content="badge test"
    size="lg"
    theme="primary"
    type="legend"
/>,
}
`;
